FileDelete/FileDelete/ProcessReference.cpp

```diff
@@ -15,11 +15,17 @@ ProcessReference::~ProcessReference() {
 	
 }
 
-NTSTATUS ProcessReference::init(size_t pid, bool attach) {
+NTSTATUS ProcessReference::init(size_t pid, bool attach)
+{
 	CHECK(PsLookupProcessByProcessId(reinterpret_cast<HANDLE>(pid), &m_process));
 	m_attach = attach;
-	if (attach) {
-		m_apc_state = (KAPC_STATE*)ExAllocatePool(NonPagedPool, sizeof(KAPC_STATE));
+	if (attach)
+	{
+		m_apc_state = (KAPC_STATE*)ExAllocatePool2(NonPagedPool, sizeof(KAPC_STATE), '2cba');
```

Comment:
ExAllocatePool2 and ExAllocatePoolZero are both functions that are supported from new versions of windows.
ExAllocatePoolWithTag is maybe deprecated in the new WDK toolset but there is background compatibility in the windows kernel, you should compile with WDK from from lower version. 
Please use ExAllocatePoolWithTag

Reply:
That's OK in an older version.